src/sagas/RootSaga.js

```diff
@@ -38,29 +38,39 @@ import { keepKeys } from '../services/utils';
 
 export function* handleFetchProfile(action) {
   const { username } = action.payload;
-  const currentUsername = yield select(state => state.authentication.username); // eslint-disable-line
+  const { isCurrentUserProfile, userAccount } = yield select(handleFetchProfileSelector);
+
+  // Default our data.
+  let preferences = {};
+  let account = userAccount;
+  let courseCertificates = null;
 
   try {
     yield put(fetchProfileBegin());
 
+    // Depending on which profile we're loading, we need to make different calls.
     const calls = [
-      call(ProfileApiService.getAccount, username),
+      // We'll always make a call for certificates.
       call(ProfileApiService.getCourseCertificates, username),
     ];
 
-    if (username === currentUsername) {
+    if (isCurrentUserProfile) {
+      // If the profile is for the current user, get their preferences.
```

Comment:
Same here.  Ideally what's happening should be clear from the code (which it is here); comments should be for when you have to add context to explain _why_ you're doing something.

Reply:
I add comments when I feel like the decisions in a piece of code are opaque to someone without any context.  In this function's case, there are specific calls we need to make in different situations, which interacts with how we're setting up the data.  I'm okay with some hand holding if it helps anyone wondering what's going on, personally.  That said, I'm updating the comments to be more specific to the decision process.